packages/editor/src/bindings/post-meta.js

```diff
@@ -51,8 +51,14 @@ function getPostMetaFields( select, context ) {
 	const registeredFields = getRegisteredPostMeta( context?.postType );
 	const metaFields = {};
 	Object.entries( registeredFields || {} ).forEach( ( [ key, props ] ) => {
-		// Don't include footnotes or private fields.
-		if ( key !== 'footnotes' && key.charAt( 0 ) !== '_' ) {
+		if (
+			// Don't include footnotes.
+			key !== 'footnotes' &&
+			// Don't include private fields.
+			key.charAt( 0 ) !== '_' &&
+			// Only support string types.
+			typeof entityMetaValues?.[ key ] === 'string'
```

Comment:
We could add the filter here:
```
const metaProperties =
				options?.schema?.properties?.meta?.properties;
			const stringMetaProperties = Object.fromEntries(
				Object.entries( metaProperties ).filter(
					( [ , value ] ) => value.type === 'string'
				)
			);
			dispatch.receiveRegisteredPostMeta(
				postType,
				stringMetaProperties
			);
```

https://github.com/WordPress/gutenberg/blob/0010f6355a37b647f2246ddbd88bba886a84a9cb/packages/core-data/src/resolvers.js#L1011-L1018

Reply:
Hmm, I think it makes more sense to have the check here in `packages/editor/src/bindings/post-meta.js`. That makes the check more specific to bindings and keeps the related exceptions together, whereas `getRegisteredPostMeta` seems to me better to me conceptually as a more agnostic function.

Reply:
I believe the conditional should look like this, adding some handling for default values, otherwise valid fields will get skipped over in templates where `entityMetaFields` is undefined due to `postId` being undefined. Consequently, this also fixes some broken tests.

It might cause other things to break, though — I haven't been able to confirm that yet:

```
if (
	// Don't include footnotes.
	key !== 'footnotes' &&
	// Don't include private fields.
	key.charAt( 0 ) !== '_' &&
	// Only support string types.
	(typeof entityMetaValues?.[ key ] === 'string' || typeof props.default === 'string')
) { // logic here }
```

Reply:
I agree `getRegisteredPostMeta` should be agnostic to bindings and in other case an object or an array could be a valid value. I want to explore if it should happen in the bindings hooks instead of just post meta.

And it's true that we can't just check against `string` because `undefined` is also a valid value. I'll spend some time today exploring the best way to handle this.